app/src/userpages/components/StreamLivePreview/index.jsx

```diff
@@ -48,7 +49,21 @@ const mapDispatchToProps = (dispatch: Function, { match: { params: { streamId }
         dispatch(openStream(streamId))
         return dispatch(getStream(streamId))
     },
-    onClose: () => dispatch(goBack()),
+    onClose: () => {
+        const { href } = window.location
+
+        dispatch(goBack())
+
+        setTimeout(() => {
+            if (window.location.href !== href) {
+                return
+            }
+
+            dispatch(push(routes.stream({
+                id: streamId,
+            })))
+        }, 100)
```

Comment:
If after 100ms location didn't change then we go to the stream page. That's a one hacky way, yup.